### ISSUE TITLE
Feature request: Block potential of code injection via macOS' dynamic linker for improved security

### ISSUE THREAD

**Issue by ghost** [localsend/localsend#2085]
### Is your feature request related to a problem? Please describe.

The user can enable a column in Activity Monitor to check for a flag called "Restricted" which is a flag that prevents programs from injecting code via macOS's [dynamic linker](https://pewpewthespells.com/blog/blocking_code_injection_on_ios_and_os_x.html). Ideally, this flag should be set but it is not for LocalSend.

### Describe the solution you'd like

For the `Restricted` flag to be set for LocalSend on macOS

### Describe alternatives you've considered

None.

**Comment by ShlomoCode** [comment]
https://docs.flutter.dev/platform-integration/macos/building#hardened-runtime

**Comment by ShlomoCode** [comment]
>The user can enable a column in Activity Monitor to check for a flag called "Restricted" which is a flag that prevents programs from injecting code via macOS's [dynamic linker](https://pewpewthespells.com/blog/blocking_code_injection_on_ios_and_os_x.html). Ideally, this flag should be set but it is not for LocalSend.

No, the Restricted flag applies only to Apple’s system apps, which are protected by SIP

**Comment by ShlomoCode** [comment]
>The user can enable a column in Activity Monitor to check for a flag called "Restricted" which is a flag that prevents programs from injecting code via macOS's [dynamic linker](https://pewpewthespells.com/blog/blocking_code_injection_on_ios_and_os_x.html). Ideally, this flag should be set but it is not for LocalSend.

No, the `Restricted` flag applies only to Apple’s system apps, which are protected by SIP.
What protects against injection is [Hardened Runtime](https://developer.apple.com/documentation/security/hardened-runtime), and it is enabled in the DMG version because the signing script uses `codesign --options runtime`.

You can run the following command to make sure Hardened Runtime is enabled for LocalSend's DMG version:

```
➜  Downloads codesign -dv --verbose=4 LocalSend.app/Contents/MacOS/LocalSend
Executable=/Users/shlomo/Downloads/LocalSend.app/Contents/MacOS/LocalSend
Identifier=org.localsend.localsendApp
Format=app bundle with Mach-O universal (x86_64 arm64)
CodeDirectory v=20500 size=2598 flags=0x10000(runtime) hashes=70+7 location=embedded
...
```

The `runtime` flag means Hardened Runtime is enabled.

**However**, since the option is not explicitly enabled in the xcodeproj, the Mac App Store version is not hardened (the flag does not exist).

Thank you for bringing this to our attention!
